Sources/MapboxCoreNavigation/EHorizon/ElectronicHorizonDelegate.swift

```diff
@@ -0,0 +1,26 @@
+import Foundation
+import MapboxNavigationNative
+
+public protocol ElectronicHorizonDelegate: class {
+
+    /**
+     Might be called multiple times when the position changes
+
+     - parameter position: Current electronic horizon position (map matched position + e-horizon tree)
+     - parameter distances: Map road object id -> RoadObjectDistanceInfo for upcoming road objects
+     */
+    func didUpdatePosition(_ position: ElectronicHorizonPosition, distances: [String : RoadObjectDistanceInfo])
```

Comment:
Is there a simpler struct-like representation that we could convert ElectronicHorizonPosition to on the developer’s behalf?

Reply:
I was aiming at full NavNative functionality support. Not sure if we want to hide some of it or make it accessible less directly. I guess it could complicate the support of such functionality.